src/main/resources/assets/carpet/scripts/draw_rc_1.sc

```diff
@@ -0,0 +1,137 @@
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow>'->['draw_prism', null, false],
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow> replace <replacement>'->['draw_prism', false]
+    },
+    'arguments'->{
+        'center'->{'type'->'pos', 'loaded'->'true'},
+        'radius'->{'type'->'int', 'suggest'->[], 'min'->0},//to avoid default suggestions
+        'replacement'->{'type'->'block'},
+        'height'->{'type'->'int', 'suggest'->[],'min'->0},
+        'orientation'->{'type'->'term', 'suggest'->['x','y','z']},
+        'pointing'->{'type'->'term','suggest'->['up','down']},
+        'hollow'->{'type'->'term','suggest'->['hollow','solid']},
+    },
+    'scope'->'global'
+};
+
+
+//"Boilerplate" code
+
+set_block(x, y, z, block, replacement)-> (
+    if(block != block(x, y, z),
+        set([x, y, z],block);
+        global_affected += 1;
+    );
+);
+
+global_affected = 0;
+
+affected(player) -> (
+    print(player,format('gi Filled ' + global_affected + ' blocks'));
+    affected = global_affected;
+    global_affected = 0;
+    affected
+);
+
+length_sq(vec) -> _vec_length(vec)^2;
+
+fill_flat(pos, offset, dr, rectangle, orientation, block, hollow, replacement)->(
+    r = floor(dr);
+    drsq = dr^2;
+    if(orientation=='x',
+        scan(pos,0,-r,-r,
+            if((!hollow && (rectangle || _y^2 + _z^2 <= drsq))||//if not hollow, vry simple
+                (hollow && ((rectangle && (abs(_y) == r || abs(_z) ==r)) || //If hollow and it's a rectangle
+                (!rectangle && (_y^2 + _z^2 <= drsq && (abs(_y)+1)^ 2 + (abs(_z)+1)^2 >= drsq)))),//If hollow and not rectangle
+                set_block(_x+offset,_y,_z,block, replacement)
+            )
+        ),
+    orientation == 'y',
+        scan(pos,-r,0,-r,
+            if((!hollow && (rectangle || _x^2 + _z^2 <= drsq))||//if not hollow, vry simple
+                (hollow && ((rectangle && (abs(_x) == r || abs(_z) ==r)) || //If hollow and it's a rectangle
+                (!rectangle && (_x^2 + _z^2 <= drsq && (abs(_x)+1)^ 2 + (abs(_z)+1)^2 >= drsq)))),//If hollow and not rectangle
+                set_block(_x,_y+offset,_z,block, replacement)
+            )
+        ),
+    orientation == 'z',
+        scan(pos,-r,-r,0,
+            if((!hollow && (rectangle || _y^2 + _x^2 <= drsq))||//if not hollow, vry simple
+                (hollow && ((rectangle && (abs(_y) == r || abs(_x) ==r)) || //If hollow and it's a rectangle
+                (!rectangle && (_y^2 + _x^2 <= drsq && (abs(_y)+1)^ 2 + (abs(_x)+1)^2 >= drsq)))),//If hollow and not rectangle
+                set_block(_x,_y,_z+offset,block, replacement)
+            )
+        ),
+        print(player(),format('r Error while running command: orientation can only be "x", "y" or "z", '+orientation+' is invalid.'));
+        global_affected = 0;
+    );
+);
+
+//drawing commands
+
+draw_sphere(centre, radius, block, replacement, hollow)->(
```

Comment:
you know that we will need to rewrite the hollow sphere from this from O(n^3) to a O(n^2) algorithm that is currently in java for draw...

Reply:
but we can do that later.

Reply:
u do realise that ur algorithm for the DrawCommand.java has O(n^3) right...

Reply:
im resarcing rasterization algorithms rn.

Reply:
true, that should use rather trig functions over angles, roughly the way I generate sphere meshes in shapes at https://github.com/gnembon/fabric-carpet/blob/7645312856c1c5970bd5b47970c07a381fc75c75/src/main/java/carpet/script/utils/ShapesRenderer.java#L851
which is definitely rendering at 2D.

Don't worry about that here - do whatever is valid and we will continue to work on it as it goes.

Reply:
![image](https://user-images.githubusercontent.com/53367549/100114508-cbc80e80-2e71-11eb-9b71-892be6da0d8b.png)

Reply:
I believe that this justifies the bruteforce code. Best I could come up with was caching already printed shapes, so u can spam same shape over and over and it's very fast.